File: functions/libgcrypt/variants/go/0_barrett_reduce_0.go
package main

func barrettReduce(a int16) int16 {
	const v int16 = ((1 << 26) + 3329/2) / 3329
	t := ((int32(v)*int32(a) + (1 << 25)) >> 26) * 3329
	return a - int16(t)
}


func main(){
    barrettReduce(0)
}

File: functions/libgcrypt/variants/c/0_barrett_reduce_0.c
#include <stdint.h>

int16_t barrett_reduce(int16_t a) {
  int16_t t;
  const int16_t v = 12482;

  t = (((int32_t)v * a) + 33554432) >> 26;
  t = t * 3329;
  return a - t;
}


int main(){
    barrett_reduce(0);
}